query-algebrizer/src/cc.rs

```diff
@@ -242,30 +265,63 @@ impl Default for ConjoiningClauses {
             aliaser: default_table_aliaser(),
             from: vec![],
             wheres: vec![],
-            bindings: BTreeMap::new(),
+            input_variables: BTreeSet::new(),
+            column_bindings: BTreeMap::new(),
+            known_bindings: BTreeMap::new(),
             known_types: BTreeMap::new(),
             extracted_types: BTreeMap::new(),
         }
     }
 }
 
 impl ConjoiningClauses {
+    fn with_bindings(bindings: BTreeMap<Variable, TypedValue>) -> ConjoiningClauses {
```

Comment:
Consider `with_known_bindings` or `with_input_values`.  You just distinguished two types of bindings; let's not muddy the waters with a function name that doesn't distinguish.